dora/core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -1004,81 +1002,42 @@ public synchronized void resetPriorities() throws IOException {
+        return String.format("<%s> is not part of the quorum <%s>.",
+            strAddr, peers.stream().map(RaftPeer::getAddress).collect(Collectors.toList()));
       }
 
       RaftPeerId newLeaderPeerId = RaftJournalUtils.getPeerId(serverAddress);
-      /* update priorities to enable transfer */
-      List<RaftPeer> peersWithNewPriorities = new ArrayList<>();
-      for (RaftPeer peer : oldPeers) {
-        peersWithNewPriorities.add(
-            RaftPeer.newBuilder(peer)
-                .setPriority(peer.getId().equals(newLeaderPeerId) ? 2 : 1)
-                .build()
-        );
-      }
-      try (RaftClient client = createClient()) {
-        String stringPeers = "[" + peersWithNewPriorities.stream().map(RaftPeer::toString)
-            .collect(Collectors.joining(", ")) + "]";
-        LOG.info("Applying new peer state before transferring leadership: {}", stringPeers);
-        RaftClientReply reply = client.admin().setConfiguration(peersWithNewPriorities);
-        processReply(reply, "failed to set master priorities before initiating election");
-      }
       /* transfer leadership */
       LOG.info("Transferring leadership to master with address <{}> and with RaftPeerId <{}>",
           serverAddress, newLeaderPeerId);
-      // fire and forget: need to immediately return as the master will shut down its RPC servers
-      // once the TransferLeadershipRequest is initiated.
-      final int SLEEP_TIME_MS = 3_000;
-      final int TRANSFER_LEADER_WAIT_MS = 30_000;
-      new Thread(() -> {
-        try (RaftClient client = createClient()) {
-          Thread.sleep(SLEEP_TIME_MS);
-          RaftClientReply reply1 = client.admin().transferLeadership(newLeaderPeerId,
-              TRANSFER_LEADER_WAIT_MS);
-          processReply(reply1, "election failed");
-        } catch (Throwable t) {
-          LOG.error("caught an error when executing transfer: {}", t.getMessage());
-          // we only allow transfers again if the transfer is unsuccessful: a success means it
-          // will soon lose primacy
-          mTransferLeaderAllowed.set(true);
-          mErrorMessages.put(transferId, TransferLeaderMessage.newBuilder()
-              .setMsg(t.getMessage()).build());
-          /* checking the transfer happens in {@link QuorumElectCommand} */
-        }
-      }).start();
-      LOG.info("Transferring leadership initiated");
+      try (RaftClient client = createClient()) {
+        RaftClientReply reply1 = client.admin().transferLeadership(newLeaderPeerId, 30_000);
+        processReply(reply1, "election failed");
+      }
+      return "";
     } catch (Throwable t) {
-      mTransferLeaderAllowed.set(true);
       LOG.warn(t.getMessage());
-      mErrorMessages.put(transferId, TransferLeaderMessage.newBuilder()
-          .setMsg(t.getMessage()).build());
+      return t.getMessage();
+    } finally {
+      mTransferLeaderAllowed.set(true);
```

Comment:
why is this set true even if an exception is thrown here?

Reply:
I've actually removed this boolean. Before, `transferLeadership` triggered the transfer asynchronously through a thread. Now, it processes the transfer synchronously. Given that `gainPrimacy`, `losePrimacy`, and `transferLeadership` are all `synchronized` methods, there is no longer a need for a boolean for synchronization.